TheOracle2/Commands/AutocompleteHandlers/CharacterAutocomplete.cs

```diff
@@ -7,32 +7,55 @@ namespace TheOracle2;
 public class CharacterAutocomplete : AutocompleteHandler
 {
     public EFContext Db { get; set; }
+    private const int BroadenSearchAt = 4;
 
     public override Task<AutocompletionResult> GenerateSuggestionsAsync(IInteractionContext context, IAutocompleteInteraction autocompleteInteraction, IParameterInfo parameter, IServiceProvider services)
     {
         try
         {
             IEnumerable<AutocompleteResult> successList = new List<AutocompleteResult>();
+            var userText = autocompleteInteraction.Data.Current.Value as string;
+            var userId = autocompleteInteraction.User.Id;
+            var guildId = context.Guild?.Id ?? autocompleteInteraction.User.Id;
 
-            var value = autocompleteInteraction.Data.Current.Value as string;
-            var guildId = (context.Guild?.Id ?? autocompleteInteraction.User.Id);
-
-            if (string.IsNullOrEmpty(value))
+            switch (userText.Length)
             {
-                var defaultChars = Db.PlayerCharacters.Where(c => c.DiscordGuildId == guildId);
-                if (defaultChars.Count() > 10) defaultChars = defaultChars.Where(c => c.UserId == context.User.Id);
-                successList = defaultChars.Select(x => new AutocompleteResult(x.Name, x.Id.ToString())).Take(SelectMenuBuilder.MaxOptionCount);
-                return Task.FromResult(AutocompletionResult.FromSuccess(successList));
+                case > 0 and < BroadenSearchAt:
+                    {
+                        // return list of guild PCs that start with query; own PCs at top.
+                        successList = Db.PlayerCharacters
+                            .Where((pc) => pc.DiscordGuildId == guildId && pc.Name.StartsWith(userText))
+                            .OrderBy(pc => pc.UserId != pc.UserId).ThenBy(pc => pc.Name)
```

Comment:
huh, weird - not sure how that crept in. it's supposed to be `userID != pc.UserId`, because `false` gets ordered before `true` (and `OrderByDescending` didn't work for some reason)

Reply:
Can you also add a comment about the hidden false sorts first functionality, and a Todo to write a custom sort method.